### ISSUE TITLE
cant create Input Layer

### ISSUE THREAD

**Issue by aligoglos** [BVLC/caffe#5438]
i run simple c++ code to pass image to my preTrained caffemodel bur when i want create network say unknown Input layer !:

Check failed: registry.count(type) == 1 (0 vs. 1) Unknown layer type: Input (known types: Convolution, Eltwise, LRN, Pooling, Power, Python, ReLU, Sigmoid, Softmax, Split, TanH)

my prototxt file :

```
#############################  DATA Layer  #############################
layer {
    name: "data_1"
    type: "Input"
    top: "data_1"
    input_param { shape: {dim:1 dim:3 dim: 640 dim: 480 } }
}
#############################  CONV NET 1 #############################
layer {
  name: "conv1_1"
  type: "Convolution"
  bottom: "data_1"
  top: "conv1_1"
  param {
    name: "conv1_w"
    lr_mult: 1
    decay_mult: 1
  }
  param {
    name: "conv1_b"
    lr_mult: 2
    decay_mult: 0
  }
  convolution_param {
    num_output: 20
    kernel_size: 4
    stride: 1
    weight_filler {
      type: "gaussian"
      std: 0.01
    }
    bias_filler {
      type: "constant"
      value: 0
    }
  }
}
layer {
  name: "relu1_1"
  type: "ReLU"
  bottom: "conv1_1"
  top: "conv1_1"
}
layer {
  name: "norm1_1"
  type: "LRN"
  bottom: "conv1_1"
  top: "norm1_1"
  lrn_param {
    local_size: 5
    alpha: 0.0001
    beta: 0.75
  }
}
layer {
  name: "pool1_1"
  type:  "Pooling"
  bottom: "norm1_1"
  top: "pool1_1"
  pooling_param {
    pool: MAX
    kernel_size: 2
    stride: 2
  }
}
layer {
  name: "conv2_1"
  type: "Convolution"
  bottom: "pool1_1"
  top: "conv2_1"
  param {
    name: "conv2_w"
    lr_mult: 1
    decay_mult: 1
  }
  param {
    name: "conv2_b"
    lr_mult: 2
    decay_mult: 0
  }
  convolution_param {
    num_output: 40
    kernel_size: 3
    group: 2
    weight_filler {
      type: "gaussian"
      std: 0.01
    }
    bias_filler {
      type: "constant"
      value: 0.1
    }
  }
}
layer {
  name: "relu2_1"
  type: "ReLU"
  bottom: "conv2_1"
  top: "conv2_1"
}
layer {
  name: "norm2_1"
  type: "LRN"
  bottom: "conv2_1"
  top: "norm2_1"
  lrn_param {
    local_size: 5
    alpha: 0.0001
    beta: 0.75
  }
}
layer {
  name: "pool2_1"
  type:  "Pooling"
  bottom: "norm2_1"
  top: "pool2_1"
  pooling_param {
    pool: MAX
    kernel_size: 2
    stride: 2
  }
}
layer {
  name: "conv3_1"
  type: "Convolution"
  bottom: "pool2_1"
  top: "conv3_1"
  param {
    name: "conv3_w"
    lr_mult: 1
    decay_mult: 1
  }
  param {
    name: "conv3_b"
    lr_mult: 2
    decay_mult: 0
  }
  convolution_param {
    num_output: 60
    kernel_size: 3
    weight_filler {
      type: "gaussian"
      std: 0.01
    }
    bias_filler {
      type: "constant"
      value: 0
    }
  }
}
layer {
  name: "pool3_1"
  type:  "Pooling"
  bottom: "conv3_1"
  top: "pool3_1"
  pooling_param {
    pool: MAX
    kernel_size: 2
    stride: 2
  }
}
layer {
  name: "conv4_1"
  type: "Convolution"
  bottom: "pool3_1"
  top: "conv4_1"
  param {
    name: "conv4_w"
    lr_mult: 1
    decay_mult: 1
  }
  param {
    name: "conv4_b"
    lr_mult: 2
    decay_mult: 0
  }
  convolution_param {
    num_output: 80
    kernel_size: 2
    stride: 2
    weight_filler {
      type: "gaussian"
      std: 0.01
    }
    bias_filler {
      type: "constant"
      value: 0.1
    }
  }
}
layer{
  name:"flatten_pool3_1"
  type:"Flatten"
  bottom:"pool3_1"
  top:"flatten_pool3_1"
}
layer{
  name:"flatten_conv4_1"
  type:"Flatten"
  bottom:"conv4_1"
  top:"flatten_conv4_1"
}
layer{
  name:"contact_conv"
  type:"Concat"
  bottom:"flatten_conv4_1"
  bottom:"flatten_pool3_1"
  top:"contact_conv"
}
layer {
  name: "deepid_1"
  type:  "InnerProduct"
  bottom: "contact_conv"
  top: "deepid_1"
  param {
    name: "fc6_w"
    lr_mult: 1
    decay_mult: 1
  }
  param {
    name: "fc6_b"
    lr_mult: 2
    decay_mult: 0
  }
  inner_product_param {
    num_output: 160
    weight_filler {
      type: "gaussian"
      std: 0.005
    }
    bias_filler {
      type: "constant"
      value: 0.1
    }
  }
}
layer {
  name: "relu6_1"
  type: "ReLU"
  bottom: "deepid_1"
  top: "deepid_1"
}
layer {
  name: "drop6_1"
  type:  "Dropout"
  bottom: "deepid_1"
  top: "deepid_1"
  dropout_param {
    dropout_ratio: 0.5
  }
}
layer {
  name: "fc8_1"
  type:  "InnerProduct"
  bottom: "deepid_1"
  top: "fc8_1"
  param {
    name: "fc8_w"
    lr_mult: 1
    decay_mult: 1
  }
  param {
    name: "fc8_b"
    lr_mult: 2
    decay_mult: 0
  }
  inner_product_param {
    num_output: 10575
    weight_filler {
      type: "gaussian"
      std: 0.01
    }
    bias_filler {
      type: "constant"
      value: 0
    }
  }


}
layer {
  name: "loss"
  type: "Softmax"
  bottom: "fc8_1"
  top: "loss"
}
```

and my simple code :

```
#include <cuda_runtime.h>
#include <cstring>
#include <cstdlib>
#include <vector>
#include <string>
#include <iostream>
#include <stdio.h>
#include "caffe/caffe.hpp"
#include "caffe/util/io.hpp"
#include "caffe/blob.hpp"

using namespace caffe;
using namespace std;

int main(int argc, char** argv) {
 Caffe::set_mode(Caffe::CPU);
 caffe::string netS = "C:/Users/127051/Documents/Visual Studio 2015/Projects/C++/Caffe/CaffeTest/x64/Release/net_struct.prototxt";
 caffe::string netW = "C:/Users/127051/Documents/Visual Studio 2015/Projects/C++/Caffe/CaffeTest/x64/Release/net_weights.caffemodel";
 //get the net
 Net<float> caffe_test_net(netS,TEST);
 //get trained net
 caffe_test_net.CopyTrainedLayersFrom(netW);

 //get datum
 Datum datum;
 if (!ReadImageToDatum("1-08.jpg", 1, 640, 480, &datum)) {
 LOG(ERROR) << "Error during file reading";
 }
 //get the blob
 Blob<float>* blob = new Blob<float>(1, datum.channels(), datum.height(), datum.width());
 //get the blobproto
 BlobProto blob_proto;
 blob_proto.set_num(1);
 blob_proto.set_channels(datum.channels());
 blob_proto.set_height(datum.height());
 blob_proto.set_width(datum.width());
 const int data_size = datum.channels() * datum.height() * datum.width();
 int size_in_datum = std::max<int>(datum.data().size(),
 datum.float_data_size());
 for (int i = 0; i < size_in_datum; ++i) {
 blob_proto.add_data(0.);
 }
 const string& data = datum.data();
 if (data.size() != 0) {
 for (int i = 0; i < size_in_datum; ++i) {
 blob_proto.set_data(i, blob_proto.data(i) + (uint8_t)data[i]);
 }
 }
 //set data into blob
 blob->FromProto(blob_proto);
 //fill the vector
 vector<Blob<float>*> bottom;
 bottom.push_back(blob);
 float type = 0.0;
 const vector<Blob<float>*>& result = caffe_test_net.Forward(bottom, &type);
 //Here I can use the argmax layer, but for now I do a simple for :)
 float max = 0;
 float max_i = 0;
 for (int i = 0; i < 1000; ++i) {
 float value = result[0]->cpu_data()[i];
 if (max < value) {
 max = value;
 max_i = i;
 }
 }
 LOG(ERROR) << "max: " << max << " i " << max_i;
 return 0;
}
```
### Your system configuration
Operating system: windows 10
Compiler: visual c++ 2015
CUDA version (if applicable): 8.0
CUDNN version (if applicable): 8.0
BLAS: OpenBLAS-v0.2.19-Win64-int32
Python or MATLAB version (for pycaffe and matcaffe respectively): python 3.5

**Comment by hgaiser** [comment]
I'm pretty sure you should post questions like this on the [caffe-users](https://groups.google.com/forum/#!forum/caffe-users) mailing list.

**Comment by williford** [comment]
... or stackoverflow.com.

**Comment by aligoglos** [comment]
my proposed [solution](http://stackoverflow.com/questions/42993870/caffe-layer-creation-failure-unknown-layer-type?noredirect=1#comment73079201_42993870)
